### ISSUE TITLE
Does not work with global packages

### ISSUE THREAD

**Issue by DanielRuf** [ds300/patch-package#116]
`yarn global add` creates an empty `node_modules` and places the dependencies one level up.
I guess the same applies to npm too.

**Comment by ds300** [comment]
Not sure if this is viable. Probably the best option is to clone the repo for the package you want to patch, and then link it, making it available globally, then patch it and save the changes in git as if it were your own project.

**Comment by ds300** [comment]
Yeah that's a viable option too if there's no build step, or if you add the build step to the package's `postinstall`.

**Comment by smashah** [comment]
Just for reference @ds300 it would be useful in a case like this:

https://github.com/Unitech/pm2/pull/5330

Where a feature is listed in the docs but not actually merged or published for people to use. Oven-ready PR waiting for 7 months with no movement. Meanwhile smaller issues are being resolved and shipped. Using the abandoned PR fork entirely will mean users miss out on recent updates. Being able to patch via patch-package could give users best of both worlds - recent versions of packages + cherry picked PRs.

I'm sure there are loads of instances out there where maintainership burden results in unmerged but ready PRs (another pops into mind https://github.com/localtunnel/server) going to "waste".

If patch-package worked on a global level then it will be incredibly useful. I can see myself making some sort of tool/bot where devs can input a package & a PR which then spits out a patch-package patch & share on the PR for others to use.

Hope I made the case to re-open this issue.

**Comment by DanielRuf** [comment]
@smashah generally speaking you can append `.patch`to any PR or commit to get the changes as pacth file and apply them via `patch` in Linux / MacOS.

See https://github.com/Unitech/pm2/pull/5330.patch and https://github.com/Unitech/pm2/commit/d6458d8536a92aa9102592e9381dc3c2fa9d1c99.patch for examples.

Basically what we do in projects is to apply the changes to a copy and make a patch file with `diff -ur` and similar. Lately we have been migrating to yarn and now to pnpm, both also support patching packages.

https://pnpm.io/cli/patch

Didn't try this with global package.

**Comment by DanielRuf** [comment]
I think it would be an option to work with the prefix or path to the global packages and run the command there with some tricks.
